File: src/sorting/selectionSort.c
#include <stdio.h>
#include <stdlib.h>
#include "./_common_/common.h"

void selection_sort(int* arr, const int size)
{
    for (int i = 0; i < size - 1; i++)
    {
        int min_idx = i;
        for (int j = i + 1; j < size; j++)
            if (arr[min_idx] > arr[j])
                min_idx = j;
        if (min_idx != i) swap(&arr[min_idx], &arr[i]);
    }
}

int main(void)
{
    int n;

    scanf("%d", &n);
    int* arr = (int*)malloc(sizeof(int) * n);
    for (int i = 0; i < n; i++)
        scanf("%d", &arr[i]);
    print_array(arr, n);
    selection_sort(arr, n);
    print_array(arr, n);

    free(arr);
    return 0;
}

/*
input:
5
5 4 3 2 1

output:
5 4 3 2 1
1 2 3 4 5
*/

File: src/sorting/selectionSort.java
import java.util.*;

class SelSort
{
    private int[] arr;
    private int size;
    
    public SelSort()
    {
        var input = new Scanner(System.in);
        size = input.nextInt();
        arr = new int[size];

        for (int i = 0; i < size; i++)
            arr[i] = input.nextInt();
        input.close();
    }

    public void selectionSort()
    {
        for (int i = 0; i < size - 1; i++)
        {
            int minIdx = i;
            for (int j = i + 1; j < size; j++)
                if (arr[minIdx] > arr[j])
                    minIdx = j;
            if (minIdx != i)
            {
                int temp = arr[i];
                arr[i] = arr[minIdx];
                arr[minIdx] = temp;
            }
        }
    }

    public void printArray()
    {
        for (int e : arr)
            System.out.printf("%d ", e);
        System.out.println();
    }
}

public class selectionSort
{
    public static void main(String[] args)
    {
        var s = new SelSort();
        s.printArray();
        s.selectionSort();
        s.printArray();
    }
}

/*
Input:
5
5 4 3 2 1

Output:
5 4 3 2 1
1 2 3 4 5
*/